src/tools/rest-frontend/resources/structure.json.in

```diff
@@ -0,0 +1,240 @@
+[
+    {
+        "name": "Documentation",
+        "type": "submenu",
+        "ref": "documentation",
+        "children": [
+            {
+                "name": "Getting started",
+                "type": "staticlist",
+                "ref": "docgettingstarted",
+                "children": [
+                    {
+                        "name": "Why Elektra?",
+                        "type": "staticfile",
+                        "options": {
+                            "path": "doc/WHY.md"
+                        }
+                    },
+                    {
+                        "name": "Big Picture",
+                        "type": "staticfile",
+                        "options": {
+                            "path": "doc/BIGPICTURE.md"
+                        }
+                    },
+                    {
+                        "name": "Compiling",
+                        "type": "staticfile",
+                        "options": {
+                            "path": "doc/COMPILE.md"
+                        }
+                    },
+                    {
+                        "name": "Installation",
+                        "type": "staticfile",
+                        "options": {
+                            "path": "doc/INSTALL.md"
+                        }
+                    },
+                    {
+                        "name": "Tutorials",
+                        "type": "staticref",
+                        "options": {
+                            "path": "tutorials"
+                        }
+                    }
+                ]
+            },
+            {
+                "name": "Tutorials",
+                "type": "parsereadme",
+                "ref": "tutorials",
+                "options": {
+                    "path": "doc/tutorials/README.md",
+                    "target_file": [],
+                    "parsing": {
+                        "stop_regex": "####### UNUSED ######",
+                        "section_regex": "## ([^#]+)",
+                        "entry_regex": "^\\- \\[(.+)\\]\\(([^\\)]+)\\)(.*)"
+                    },
+                    "name": {
+                        "make_pretty": false
+                    }
+                }
+            },
+            {
+                "name": "Plugins",
+                "type": "parsereadme",
+                "ref": "plugins",
+                "options": {
+                    "path": "src/plugins/README.md",
+                    "target_file": ["README.md", "README", "readme.md", "readme"],
+                    "parsing": {
+                        "start_regex": "# Plugins #",
+                        "stop_regex": "####### UNUSED ######",
+                        "section_regex": "### ([^#]+) ###",
+                        "entry_regex": "^\\- \\[(.+)\\]\\(([^\\)]+)\\)(.*)"
+                    },
+                    "name": {
+                        "make_pretty": false
```

Comment:
It will make the names capitalized and other things. As the plugins have pretty names anyway (no special characters or whatever), I think it is best to keep them naturally.

I added the feature for this structure type because it was easy to do, but normally you would only use it for the "listfiles" structure type for example, when you want to get rid of names like "application-integration.md" by transforming them into "Application Integration". The "parsereadme" type assumes that the parsed text is already pretty (it is in a readme, so it should be).